lib/assessment.js

```diff
@@ -39,7 +49,12 @@ var newArray2 = function newArray2(length, defaultsFunction) {
 // The function you write should return
 // the object passed in as an argument or the object created
 var addMethod = function addMethod(propertyName, method, obj) {
-
+  if (obj){
+  obj[propertyName] = method;
+  }else{
+    obj = {};
+    obj[propertyName] = method;
+  }
 };
```

Comment:
All the side effects here are correct, but you forgot to `return` the object passed in or created. If we used this function to create an object, its reference would be lost, and it would be immediately discarded. 3/4 credit.